### PR TITLE
openmpi: Declare more build dependencies for `@master`

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

**Comment by eschnett** [comment]
CI failed because one step in the pipeline was cancelled. I think we should re-run it.

**Comment by haampie** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 1]
```diff
@@ -321,6 +321,11 @@ class Openmpi(AutotoolsPackage):
 
     depends_on('openssh', type='run')
 
+    depends_on('autoconf @2.69:', type='build', when='@master')
+    depends_on('automake @1.13.4:', type='build', when='@master')
+    depends_on('libtool @2.4.2:', type='build', when='@master')
+    depends_on('pandoc', type='build', when='@master')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/openmpi/package.py]: Can we modify the directives at line 275 instead of adding new ones?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/openmpi/package.py (hunk 1 above)
Comment (eschnett) [reply var/spack/repos/builtin/packages/openmpi/package.py]: Done.

**Comment by alalazo** [comment]
Can you rebase to run Gitlab pipelines on a new merge commit?

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
No issues with Spack idioms, but I'll be waiting to hear from @hppritcha before merging

**Comment by eschnett** [comment]
@hppritcha do you want to comment?

**Comment by hppritcha** [comment]
hmmm I guess thumbs up emoji no good. 
Yes this PR looks okay to me.

**Review by alalazo — APPROVED** [review]

**Pull request merged**